src/nvim/mod.rs

```diff
@@ -96,6 +110,9 @@ impl fmt::Display for NvimInitError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             Self::ResponseError { source, .. } => write!(f, "{source:?}"),
+            Self::TcpConnectError { source, addr } => {
+                write!(f, "Failed to connect to {addr}: {source:?}")
+            }
             Self::MissingCapability(cap) => {
                 write!(f, "Nvim version is too old, missing support for {cap}")
             }
```

Comment:
You also might want to update `error::Error::cause()` down below ↓

Reply:
used the same implementation as for `ResponseError`